### ISSUE TITLE
ActiveRecord::UnknownAttributeError when using AR built-in methods

### ISSUE THREAD

**Issue by iMacTia** [czaks/detailed#1]
Hi there,

first of all, THANKS!
I've been looking around for a solution to implement MTI in Rails and this looks to me as the most interesting. I'm still not 100% sure it fits my case, but I decided to give it a try and eventually contribute if something is missing. Please, don't let it die and keep updating it!

Going to the problem...
Following the instructions, the gem works as expected, I can in fact do something like:

```
sc = SubClass.new
sc.field_in_the_detail_table = 'some_value'
sc.save
```

I tried to do the same using Active Record built-in methods like this:

```
SubClass.create(field_in_the_detail_table: 'some_value')
```

But this actually raise the exception `ActiveRecord::UnknownAttributeError`. That's probably because fields from the detail table are directly accessible through the method_missing trick, and the self.details instance, but the correspondent `after_initialize` bit of code is called AFTER AR try to set the attributes, resulting in the error shown.

Any clue on how to solve this?

Mattia

**Comment by czaks** [comment]
Ok, I see that you solved it now with #2 

The AssociationScope was a hack to allow this: https://github.com/czaks/detailed#subclasses-associations . 

I don't use the gem anymore, as the project that used it is now unmaintained and I don't have any Rails 4 project to test it out. If you want to improve it, feel free, I gave you commit rights :)

**Comment by iMacTia** [comment]
What a pity! I think this gem have a huge potential!

Because I'm going to use it in my project, I'll see if I can add something in the meantime.
Everything will be tested under Rails 4.2, so maybe we should create a new branch or at least push to a new version (I left 0.1.1 in my commit).
I understand the aim of the Extension, however the method `add_constraints` changed in Rails 4 and I think I'll need some time to figure out how to make it working there.

Hope to have your thoughts about new changes!

Thanks

**Comment by iMacTia** [comment]
Fixed with #2
